gravitee-apim-gateway/gravitee-apim-gateway-tests-sdk/src/main/java/io/gravitee/apim/gateway/tests/sdk/AbstractGatewayTest.java

```diff
@@ -269,6 +270,27 @@ private void updateEndpoints(ReactableApi<?> reactableApi) {
                 }
             }
         }
+
+        if (reactableApi.getDefinition() instanceof io.gravitee.definition.model.v4.Api) {
+            var api = (io.gravitee.definition.model.v4.Api) reactableApi.getDefinition();
+            api
+                .getEndpointGroups()
+                .stream()
+                .flatMap(eg -> eg.getEndpoints().stream())
+                .filter(endpoint -> endpoint.getType().equals("http-proxy"))
+                .forEach(
+                    endpoint -> {
+                        var port = endpoint.getConfiguration().contains("https") ? wiremockHttpsPort : wiremockPort;
+
+                        endpoint.setConfiguration(endpoint.getConfiguration().replace("8080", Integer.toString(port)));
+                    }
+                );
+
+            // Workaround to re-deploy the api to update endpoint config
+            var manager = applicationContext.getBean(ApiManager.class);
+            manager.unregister(reactableApi.getId());
+            manager.register(reactableApi);
```

Comment:
Not sure to have a good understanding on why this hack is only necessary for v4 api and not v2 ones ?
@ytvnr do you have any suggestion regarding that point ?

Reply:
in v2, Gateway relies directly on the `(Reactable)Api` object to use the endpoints, meaning modifying the endpoint of the `ReactableApi` will be take into consideration at request time.

For the v4, I suppose that since endpoint is a connector, it is instantiated at deployment time (as for policies), meaning the configuration cannot be changed after deployment.

@jhaeyaert, does that make sense ?

@jgiovaresco, I would suggest to create a new method for this case and call it the same way `configureApi()` or `ensureMinimalRequirementForApi()` is called (before deployment in `GatewayRunner#deploy`)